Amplitude/Amplitude.m

```diff
@@ -257,6 +277,15 @@ - (void)initializeApiKey:(NSString*) apiKey
 
 - (void)initializeApiKey:(NSString*) apiKey userId:(NSString*) userId
 {
+    UIApplicationState state = [UIApplication sharedApplication].applicationState;
+    [self initializeApiKey:apiKey userId:userId startSession:(state != UIApplicationStateBackground)];
+}
```

Comment:
There are a few cases when the app runs in the background and if the SDK is initialized for tracking during those times, no session would be started. Pretty sure we don't want a session in those cases, though. And, later, if the app ever becomes active, the UIApplicationEnteredForeground handler would start a session correctly.

There is a class of apps where the user 'interacts' with the app while it's in the background and you'd want to have a session -- background audio players, etc. This change won't cause anything different in this case. The developer probably isn't initializing Amplitude in the "next track" handler.

Also, the new notification actions introduced in iOS8 could pose an issue. The application never becomes active, but the user has explicitly interacted. For example, a user receives a notification for an email message and they swipe over and hit delete from the lock screen.

Probably these two cases are lightweight enough that we won't want a user session logged anyway. I don't see a way to be smart enough to get this right in all cases automatically. We should expose startSession publicly so that developers can override that if they want a session in one of those cases in places where they're not initializing.